app/frontend/src/components/MapSearch.tsx

```diff
@@ -78,13 +81,17 @@ export default function MapSearch({
       },
     };
     try {
-      const res = await fetch(url, options);
-      const data = (await res.json()) as Array<any>;
+      const response = await fetch(url, options);
+      const nominatimResults = (await response.json()) as Array<NominatimPlace>;
+
       setSearchOptions(
-        data.map((obj) => ({
-          name: obj["display_name"],
-          location: new LngLat(Number(obj["lon"]), Number(obj["lat"])),
-        }))
+        nominatimResults.map((result) => {
+          return {
+            name: result["display_name"],
+            simplifiedName: simplifyPlaceDisplayName(result),
```

Comment:
This is much better! I would be up for just ditching the full name and using only `name: simplifyPlaceDisplayName(result)` so the list of options has the better-looking name too. Up to you!

Reply:
Wanted to do that but the problem with this approach is that nominatim is still doing search on fields we are not including. The result could be e.g. displaying "Sela" after typing "fa" because there is railway containing "fa". It could be missleading for users. There is a`zoom` query param which we can use to narrow search (https://nominatim.openstreetmap.org/ui/reverse.html?format=html&lat=51.484622236435&lon=8.337727189064&zoom=14) but still there would be cases that there is something we don't want in options list.
![image](https://user-images.githubusercontent.com/10603901/107438050-35ca8880-6b30-11eb-8cf1-a21157f75b06.png)

Reply:
Or maybe we can do filtering on our end?

Reply:
In the vue frontend, the autocomplete/search list was raw nominatim output, but then in the automatically generated "display address" box we'd put the culled one

Reply:
I see, perhaps leave this then